config/pmd-test.xml

```diff
@@ -86,19 +86,10 @@
             <!-- in AstRegressionTest pmd does not find asserts in inner classes methods called from the test method -->
             <!-- in AllChecksTest pmd does not find asserts in lambdas -->
             <!-- in DetailASTTest pmd does not find asserts in lambdas -->
-            <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='SuppressionFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCode']
-            | //ClassOrInterfaceDeclaration[@Image='SuppressionCommentFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCodeOfTagClass']
-            | //ClassOrInterfaceDeclaration[@Image='SuppressWithNearbyCommentFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCodeOfTagClass']
-            | //ClassOrInterfaceDeclaration[@Image='SuppressionXpathFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCode']
-            | //ClassOrInterfaceDeclaration[@Image='XpathFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCode']
-            | //ClassOrInterfaceDeclaration[@Image='AbstractJavadocCheckTest']//MethodDeclarator[@Image='testBlockCommentPositionHasPrivateConstr']
-            | //ClassOrInterfaceDeclaration[@Image='ImportControlCheckTest']//MethodDeclarator[@Image='testPkgRegExpInChild']
-            | //ClassOrInterfaceDeclaration[@Image='ImportControlCheckTest']//MethodDeclarator[@Image='testPkgRegExpInParent']
-            | //ClassOrInterfaceDeclaration[@Image='ImportControlCheckTest']//MethodDeclarator[@Image='testPkgRegExpInBoth']
-            | //ClassOrInterfaceDeclaration[@Image='AstRegressionTest']//MethodDeclarator[@Image='testImpossibleValid']
-            | //ClassOrInterfaceDeclaration[@Image='AllChecksTest']//MethodDeclarator[@Image='testAllModulesAreReferencedInConfigFile']
-            | //ClassOrInterfaceDeclaration[@Image='DetailASTTest']//MethodDeclarator[@Image='testTreeStructure']
-            | //ClassOrInterfaceDeclaration[@Image='SuppressWithPlainTextCommentFilterTest']//MethodDeclarator[@Image='testEqualsAndHashCodeOfTagClass']
+            <property name="violationSuppressXPath" value="//PrimaryPrefix/Name[@Image='verify' or @Image='exit.checkAssertionAfterwards']
+            | //PrimarySuffix[@Image='verify']
+            | //ClassOrInterfaceDeclaration[@Image='PackageObjectFactoryTest']//MethodDeclarator[@Image='testCreateModuleWithNonExistName']
+            | //ClassOrInterfaceDeclaration[@Image='XdocsPagesTest']//MethodDeclarator[@Image='testAllChecksPresentOnAvailableChecksPage']
```

Comment:
Did you just optimize suppression expression ?
Comment above explains  suppression, now might be unsynced

Reply:
The problem is still the same: asserts in lambdas are invisible to PMD.

Reply:
> The expression changed to "any method with a call to method verify(...) or exit.checkAssertionAfterwards(...) or .verify(...).
PackageObjectFactoryTest and XdocsPagesTest do not use any of them, so they are explicit here.

ok